### PR TITLE
chore(governance): fix errors when creating Gitpod environment

### LINKED ISSUE #4530: Maintenance: Fix error when creating a developer environment in Gitpod
### Why is this needed?

When trying to run `make dev-gitpod` we get this error

```sh
- Installing poetry-bumpversion (0.0.0 315fe33): Failed

  ChefInstallError

  Failed to install poetry-core.
  
  Output:
  Updating dependencies
  Resolving dependencies...
  
  Package operations: 1 install, 0 updates, 0 removals
  
    - Installing poetry-core (1.9.0)
  
    CalledProcessError
  
    Command '['/tmp/tmp2m752m6b/.venv/bin/python', '-I', '-W', 'ignore', '-c', '\nimport importlib.util\nimport json\nimport sys\n\nfrom pathlib import Path\n\nspec = importlib.util.spec_from_file_location(\n    "packaging", Path(r"/workspace/.pyenv_mirror/user/3.9.13/lib/python3.9/site-packages/packaging/__init__.py")\n)\npackaging = importlib.util.module_from_spec(spec)\nsys.modules[spec.name] = packaging\n\nspec = importlib.util.spec_from_file_location(\n    "packaging.tags", Path(r"/workspace/.pyenv_mirror/user/3.9.13/lib/python3.9/site-packages/packaging/tags.py")\n)\npackaging_tags = importlib.util.module_from_spec(spec)\nspec.loader.exec_module(packaging_tags)\n\nprint(\n    json.dumps([(t.interpreter, t.abi, t.platform) for t in packaging_tags.sys_tags()])\n)\n']' returned non-zero exit status 1.
  
    at ~/.pyenv/versions/3.9.13/lib/python3.9/subprocess.py:528 in run
         524│             # We don't call process.wait() as .__exit__ does that for us.
         525│             raise
         526│         retcode = process.poll()
         527│         if check and retcode:
      →  528│             raise CalledProcessError(retcode, process.args,
         529│                                      output=stdout, stderr=stderr)
         530│     return CompletedProcess(process.args, retcode, stdout, stderr)
         531│ 
         532│ 
  
  The following error occurred when trying to handle this error:
  
  
    EnvCommandError
  
    Command ['/tmp/tmp2m752m6b/.venv/bin/python', '-I', '-W', 'ignore', '-c', '\nimport importlib.util\nimport json\nimport sys\n\nfrom pathlib import Path\n\nspec = importlib.util.spec_from_file_location(\n    "packaging", Path(r"/workspace/.pyenv_mirror/user/3.9.13/lib/python3.9/site-packages/packaging/__init__.py")\n)\npackaging = importlib.util.module_from_spec(spec)\nsys.modules[spec.name] = packaging\n\nspec = importlib.util.spec_from_file_location(\n    "packaging.tags", Path(r"/workspace/.pyenv_mirror/user/3.9.13/lib/python3.9/site-packages/packaging/tags.py")\n)\npackaging_tags = importlib.util.module_from_spec(spec)\nspec.loader.exec_module(packaging_tags)\n\nprint(\n    json.dumps([(t.interpreter, t.abi, t.platform) for t in packaging_tags.sys_tags()])\n)\n'] errored with the following return code 1
    
    Error output:
    Traceback (most recent call last):
      File "<string>", line 18, in <module>
      File "<frozen importlib._bootstrap_external>", line 846, in exec_module
      File "<frozen importlib._bootstrap_external>", line 982, in get_code
      File "<frozen importlib._bootstrap_external>", line 1039, in get_data
    FileNotFoundError: [Errno 2] No such file or directory: '/workspace/.pyenv_mirror/user/3.9.13/lib/python3.9/site-packages/packaging/tags.py'
    
  
    at ~/.pyenv/versions/3.9.13/lib/python3.9/site-packages/poetry/utils/env/base_env.py:342 in _run
        338│                 output = subprocess.check_output(
        339│                     cmd, stderr=stderr, env=env, text=True, **kwargs
        340│                 )
        341│         except CalledProcessError as e:
      → 342│             raise EnvCommandError(e)
        343│ 
        344│         return output
        345│ 
        346│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:
  
  Cannot install poetry-core.
  
  
  
  Error:
  

  at ~/.pyenv/versions/3.9.13/lib/python3.9/site-packages/poetry/installation/chef.py:102 in install
       98│             InstalledRepository.load(self._env),
       99│         )
      100│         installer.update(True)
      101│         if installer.run() != 0:
    → 102│             raise ChefInstallError(requirements, io.fetch_output(), io.fetch_error())
      103│ 
      104│ 
      105│ class Chef:
      106│     def __init__(

Cannot install build-system.requires for poetry-bumpversion.

make[1]: *** [Makefile:117: dev-version-plugin] Error 1
make[1]: Leaving directory '/workspace/powertools-lambda-python'
make: *** [Makefile:10: dev] Error 2
gitpod /workspace/powertools-lambda-python (develop) $ 
```

### Which area does this relate to?

Automation

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #4530 

### Changes

This PR resolves the issue of Gitpod environment configuration by removing the `poetry-bumpversion` plugin from the initial setup. This plugin is not necessary in a development environment, as the only way we perform library versioning is through our CI/CD pipeline.

Additionally, I have taken the opportunity to update the environment to use Python 3.11.

### User experience

image

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .gitpod.Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# See here all gitpod images available: https://hub.docker.com/r/gitpod/workspace-python-3.9/tags
-# Current python version: 3.9.13
-FROM gitpod/workspace-python-3.9@sha256:de87d4ebffe8daab2e8fef96ec20497ae4f39e8dcb9dec1483d0be61ea78e8cd
+# See here all gitpod images available: https://hub.docker.com/r/gitpod/workspace-python-3.11/tags
+# Current python version: 3.11.9
```
Comment (heitorlessa) [inline .gitpod.Dockerfile:2]: worth moving to 3.12?

**Review by heitorlessa — APPROVED** [review]
should we perhaps use 3.12 (latest) instead of 3.11 given the opportunity?

**Comment by leandrodamascena** [comment]
> should we perhaps use 3.12 (latest) instead of 3.11 given the opportunity?

Yeah, I tried that but for some reason the gitpod workspace for python3.12 is breaking some dependencies. I need to investigate more, but I think it's ok to use 3.11 for now. WDYT?

**Comment by heitorlessa** [comment]
Let’s go with 3.11 then.

**Pull request merged**